net.go

```diff
@@ -686,12 +686,17 @@ func (m *Memberlist) rawSendMsgPacket(addr string, node *Node, msg []byte) error
 
 	// Add a CRC to the end of the payload if the recipient understands
 	// ProtocolVersion >= 5
-	if node != nil && node.PMax >= 5 {
-		crc := crc32.ChecksumIEEE(msg)
-		header := make([]byte, 5, 5+len(msg))
-		header[0] = byte(hasCrcMsg)
-		binary.BigEndian.PutUint32(header[1:], crc)
-		msg = append(header, msg...)
+	if node != nil {
+		m.nodeLock.RLock()
+		pMax := node.PMax
+		m.nodeLock.RUnlock()
```

Comment:
Is this correct? node.PMax value may become different right after this unlock, so you may have a stale value in pMax?

Reply:
That is fine, protocol is usually defined in the config file and never change during serf. Of course someone can find a way to updated it, but it is not a concern. 

I want the fix to be safe, so try to lock as less code as possible

Reply:
don't see anything that is blocking in this function - so just do a defer RUnlock()